portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -2,24 +2,57 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import com.google.appengine.api.datastore.Entity;
 
 /**
- * Class representing a comment
+ * Class representing a comment. This class is immutable and threadsafe.
  *
  * <p>Note: The private variables in this class are converted into JSON.
  */
 public class Comment {
 
+  private final long ROOT_COMMENT_ID = 0;
   private final long id;
   private final String text;
   private final long createdAt;
   private final String email;
+  private final long parent;
+  private final long root;
 
-  public Comment(long id, String text, long createdAt, String email) {
+  public Comment(long id, String text, long createdAt, String email, long parent, long root) {
     this.id = id;
     this.text = text;
     this.createdAt = createdAt;
     this.email = email;
+    this.parent = parent;
+    this.root = root;
   }
 
+  public Comment(Entity entity) {
+    this.id = entity.getKey().getId();
+    this.text = (String) entity.getProperty("text");
+    this.email = (String) entity.getProperty("creatorEmail");
+    this.createdAt = (long) entity.getProperty("createdAt");
+    this.parent = (long) entity.getProperty("parent");
+    this.root = (long) entity.getProperty("root");
+  }
+
+  //This constructor is associated with the creation of a dummy, root level comment (a supernode for the comment tree)
+  public Comment() {
+    this.id = ROOT_COMMENT_ID;
```

Comment:
Consider calling the other `Comment(long id, String text, long createdAt, String email, long parent, long root)` constructor.